validation/src/test/scala/hmda/validation/rules/lar/validity/V505Spec.scala

```diff
@@ -0,0 +1,31 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.rules.EditCheck
+import hmda.validation.rules.lar.LarEditCheckSpec
+import org.scalacheck.Gen
+
+class V505Spec extends LarEditCheckSpec with BadValueUtils {
+  property("When action taken not in 2-8 lar is valid") {
+    forAll(larGen, intOutsideRange(2, 8)) { (lar, actionTaken) =>
+      val validLar = lar.copy(actionTakenType = actionTaken)
+      validLar.mustPass
+    }
+  }
+
+  property("When rate spread is NA lar is valid") {
+    forAll(larGen) { lar =>
+      val validLar = lar.copy(rateSpread = "NA")
+      validLar.mustPass
+    }
+  }
+
+  property("When action taken in 2-8 and ratespread not NA lar is invalid") {
+    forAll(larGen, Gen.choose(2, 8), Gen.numStr) { (lar, actionTaken, rate) =>
```

Comment:
There's a `rateSpreadGen` available, since it's defined in the `LarGenerators` file. You could use that to generate realistic rate spreads. (You'd have to use a `whenever` clause to filter out `"NA"`s)

After that, this PR LGTM.